### PR TITLE
Add binary operation lowerings.

### PR DESCRIPTION
**Pull request by vext01**

### CODE REVIEW

File: src/librustc/sir.rs [hunk 1]
```diff
@@ -200,4 +213,25 @@ impl SirFuncCx {
...
+    ) -> ykpack::Rvalue {
+        let sir_op = binop_lowerings!(
+            op, Add, Sub, Mul, Div, Rem, BitXor, BitAnd, BitOr, Shl, Shr, Eq, Lt, Le, Ne, Ge, Gt,
+            Offset
+        );
+        let sir_opnd1 = self.lower_operand(opnd1);
+        let sir_opnd2 = self.lower_operand(opnd2);
+
+        if checked {
+            ykpack::Rvalue::CheckedBinaryOp(sir_op, sir_opnd1, sir_opnd2)
```
Comment (bjorn3) [inline src/librustc/sir.rs:240]: FYI only `Add`, `Sub`, `Mul`, `Shl` and `Shr` can be checked.

Comment (vext01) [reply src/librustc/sir.rs:240]: 87f32fd adds an assertion. Do you think that's good enough?

**Comment by vext01** [comment]
I think this is ready to go.

**Comment by ltratt** [comment]
As a general point (not really for this PR), is it a bit verbose if so many functions start with `lower_`? In my experience, a shorter prefix is more sustainable in the long term (e.g. `l_`).

**Comment by vext01** [comment]
I prefer `lower_` I'm afraid.

**Comment by vext01** [comment]
Let me expand on this, as I do feel somewhat strongly.

 * I can't count the number of times I've had to ask what the meaning of a single- (or two-) letter prefix is. I'm in the position where I can easily ask what they mean, but not everyone has that luxury. It's better if the code speaks for itself, leaving no ambiguity.

 * It's not the Rust way and the precedent in the compiler is to use descriptive names, e.g. `codegen_block` instead of `c_block`.

**Comment by ltratt** [comment]
We've agree offline to leave the naming convention as-is in this repository.

**Comment by ptersilie** [comment]
Looks fine. Please squash.

As for the `lowering_` discussion, I'm leaning towards Edd here. It's still short enough to not offend my eyes. If the function name was super longer then that would need another discussion.

**Comment by vext01** [comment]
splat.

**Pull request merged**